video/graphics.h

```diff
@@ -578,6 +588,9 @@ void plotCharacter(char c) {
 	if (ttxtMode) {
 		ttxt_instance.draw_char(activeCursor->X, activeCursor->Y, c);
 	} else {
+		if (cursorBehaviour.scrollProtect) {
+			cursorAutoNewline();
+		}
```

Comment:
ensures that if we're doing scroll protect that we will deal with any pending newlines